[main] Add support for arm platform

ayakael opened the pull request:
## Expected behavior
As prebuilt SDKs are provided for arm by Microsoft, source-build should support ARM build.

## Actual behavior
Per https://github.com/dotnet/source-build/issues/2781, source-build does not support arm. 

## Proposed modifications
This pull request modifies existing `Platform` and `Architecture` logics to parse arch information correctly when `BuildArchitecture=arm`

## Varia
release/6.0.1xx version: https://github.com/dotnet/installer/pull/13378

Made as part of Alpine Linux dotnet6 packaging project, see https://github.com/dotnet/source-build/issues/2782

ayakael commented:
Rebased to main to see if checks fail here as well.

kasperk81 commented on Directory.Build.props:
```diff
@@ -9,7 +9,7 @@
   <PropertyGroup>
     <BuildArchitecture>$([System.Runtime.InteropServices.RuntimeInformation]::ProcessArchitecture.ToString().ToLowerInvariant())</BuildArchitecture>
-    <Architecture Condition="'$(Architecture)' == '' AND '$(BuildArchitecture)' == 'arm64'">$(BuildArchitecture)</Architecture>
+    <Architecture Condition="'$(Architecture)' == '' AND ('$(BuildArchitecture)' == 'arm64' OR '$(BuildArchitecture)' == 'arm')">$(BuildArchitecture)</Architecture>
     <Architecture Condition="'$(Architecture)' == '' AND '$(BuildArchitecture)' == 's390x'">$(BuildArchitecture)</Architecture>
     <Architecture Condition="'$(Architecture)' == ''">x64</Architecture>
```
```suggestion
    <Architecture Condition="'$(Architecture)' == ''">$(BuildArchitecture)</Architecture>
```

MichaelSimons commented:
> @MichaelSimons Who can I ping to review this? It'd be nice to include this in rc2 and next verson of dotnet6.

@dsplaisted - Can you please review else add the appropriate code reviewer?

dsplaisted reviewed (approved):
As far as I can tell, this looks good.

ayakael commented:
* Fixed a tab-stop typo
* squashed commits
* src/SourceBuild/tarball/content/repos/known-good.proj: per proposal in https://github.com/dotnet/installer/pull/13378

ayakael commented:
Per @omajid in https://github.com/dotnet/installer/pull/13378, reverted proposal by @kasperk81

MichaelSimons commented:
@crummel - please review.

crummel commented:
I rebased this on latest main and I'm reviewing now.
